src/Tokenizer/Analyzer/Analysis/TypeAnalysis.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tokenizer\Analyzer\Analysis;
+
+class TypeAnalysis implements StartEndTokenAwareAnalysis
```

Comment:
let us keep all analysis final+internal for the first iteration. let us keep them a bit and let them become mature before putting a BC promise on them